### PR TITLE
.github: Add Windows GPU workflow

### PR DESCRIPTION
**Pull request by seemethere**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #58880 DO NOT MERGE, testing windows GPU workflows
* #58879 .github: Enable Windows GPU workflows on PR
* **#58782 .github: Add Windows GPU workflow**

[skip ci]

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D28645531](https://our.internmc.facebook.com/intern/diff/D28645531)

### CODE REVIEW

**Comment by seemethere** [comment]
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28645531).

**Comment by seemethere** [comment]
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28645531).

**Comment by seemethere** [comment]
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28645531).

**Comment by seemethere** [comment]
Tested here: https://github.com/pytorch/pytorch/pull/58880

File: .jenkins/pytorch/win-test.sh [hunk 1]
```diff
@@ -55,7 +55,7 @@ fi
 
 run_tests() {
     # Run nvidia-smi if available
-    for path in  /c/Program Files/NVIDIA Corporation/NVSMI/nvidia-smi.exe /c/Windows/System32/nvidia-smi.exe; do
+    for path in '/c/Program Files/NVIDIA Corporation/NVSMI/nvidia-smi.exe' /c/Windows/System32/nvidia-smi.exe; do
```
Comment (samestep) [inline .jenkins/pytorch/win-test.sh:58]: What is the reason for this change?

**Review by samestep — APPROVED** [review]
lgtm

File: .jenkins/pytorch/win-test.sh (hunk 1 above)
Comment (seemethere) [reply .jenkins/pytorch/win-test.sh:58]: `/c/Program Files/NVIDIA Corporation/NVSMI/nvidia-smi.exe` without the quotes represents three arguments in the for loop:

* `/c/Program`
* `Files/NVIDIA`
* `Corporation/NVSMI/nvidia-smi.exe`

Which is probably not what we're looking for since we're looking specifically for the nvidia-smi.exe binary

**Review by malfet — APPROVED** [review]

File: .jenkins/pytorch/win-test.sh [hunk 2]
```diff
@@ -55,8 +55,8 @@ fi
 
 run_tests() {
     # Run nvidia-smi if available
-    for path in  /c/Program Files/NVIDIA Corporation/NVSMI/nvidia-smi.exe /c/Windows/System32/nvidia-smi.exe; do
-        if [ -x $path ]; then
+    for path in '/c/Program Files/NVIDIA Corporation/NVSMI/nvidia-smi.exe' /c/Windows/System32/nvidia-smi.exe; do
+        if [ -x "$path" ]; then
```
Comment (seemethere) [inline .jenkins/pytorch/win-test.sh:59]: Needed for shellcheck

**Comment by seemethere** [comment]
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28645531).

**Comment by seemethere** [comment]
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28645531).
